src/psyclone/psyir/backend/fortran.py

```diff
@@ -1350,12 +1350,27 @@ def literal_node(self, node):
                         )
                 quote_symbol = '"'
             result = f"{quote_symbol}{node.value}{quote_symbol}"
-        elif (node.datatype.intrinsic == ScalarType.Intrinsic.REAL and
-              precision == ScalarType.Precision.DOUBLE):
-            # The PSyIR stores real scalar values using the standard 'e'
-            # notation. If the scalar is in fact double precision then this
-            # 'e' must be replaced by 'd' for Fortran.
-            result = node.value.replace("e", "d", 1)
+        elif node.datatype.intrinsic == ScalarType.Intrinsic.REAL:
+            if precision == ScalarType.Precision.DOUBLE:
+                # The PSyIR stores real scalar values using the standard 'e'
+                # notation. If the scalar is in fact double precision then this
+                # 'e' must be replaced by 'd' for Fortran.
+                result = node.value.replace("e", "d", 1)
+                # Ensure it ends with ".0" if it isn't already explicitly
+                # formatted as a real.
+                try:
+                    _ = int(result)
+                    result = result + ".0d0"
```

Comment:
Is there a reason we force the specification of the real output in the backend? If its just for code appearance thats ok - but then I think this should also be updated in the other language writers as well (i think mainly `c.py` should also have this change). Alternatively we could move this logic into Literal itself

Reply:
That made me think but then I realised that this particular line is Fortran specific (the "0d0") so it does belong here rather than in the Node. I'll take a look at `c.py` as that will probably inform my opinion.

Reply:
`c.py` had a redundant 'replace "d" with "e"' (redundant because the PSyIR always uses 'e' internally) so I've tidied it up.